src/app_engine/test_utilities.py

```diff
@@ -42,13 +42,19 @@ def tearDown(self):
     self.testbed.deactivate()
     analytics.report_event = self.oldReportEvent
 
-  def checkInvalidRequests(self, path, params):
+  def checkInvalidRequests(self, path, params, jsonResult=False):
```

Comment:
I modeled the new endpoints and my updates on the join page, which returns json. You are right that things are not consistent. I think everything should return json with a result field. Otherwise clients would have to try to parse the result as json, if it fails, try to parse it as a result string. Seems like we can standardize on a message format.